src/System.Net.HttpListener/tests/HttpListenerFactory.cs

```diff
@@ -7,15 +7,34 @@
 using System.Net.Sockets;
 using System.Runtime.InteropServices;
 using System.Text;
+using System.IO;
 
 namespace System.Net.Tests
 {
     // Utilities for generating URL prefixes for HttpListener
     public class HttpListenerFactory : IDisposable
     {
-        const int MinPort = 1025;
-        private static readonly object s_createListenerLock = new object();
-        private static int s_nextPort = MinPort;
+        const int MaxStartAttempts = 50;
+        const string ProcPortRangeFileName = "/proc/sys/net/ipv4/ip_local_port_range";
+        private static int _minPort;
+        private static int _maxPort;
+        private static readonly Random _random = new Random();
```

Comment:
Nit: s_ prefix on statics.  And the s_minPort and s_maxPort could be readonly as well, right?